### PR TITLE
#267 Author should be able to delete a blog

### LINKED ISSUE #267: Author should be able to delete a blog
As the Author of the blog, I should be able to delete the blog i created.

### PR DESCRIPTION
**Pull request by mutsinziisaac**
### PR Description

This PR adds functionality to allow users to delete their blogs and updates the relevant files to support this feature.

### Description of tasks that were expected to be completed

Implement the ability for users to delete their own blogs.
Update backend routes and controllers to handle blog deletion securely and efficiently.
Modify the frontend to include a delete option for blog posts.
Ensure proper validation and authorization checks to prevent unauthorized deletions.
Previously, there was no option for users to delete their blogs, leading to an inability to manage or remove outdated or unwanted content.

### How has this been tested?

Clone the repository.
Checkout the branch delete-blog-feature.
Set up the environment variables as indicated in .env.example.
Run npm install followed by npm run dev to start the app.
Test the delete functionality on the blog list page and verify changes in the database.

### Deployed Link:
https://atlp-devpulse-fn-git-ft-delete-blog-devpulses-projects-62c56cf1.vercel.app/#/home

### Track PR (Issue number & link):

#267 

### ScreenShoot

Screenshot 2024-11-26 at 17 35 59

Screenshot 2024-11-27 at 13 07 47

### CODE REVIEW

**Review by Mugisha146 — APPROVED** [review]
Well done

**Review by Jadowacu1 — APPROVED** [review]

File: src/pages/Blogs/singleBlog.tsx [hunk 1]
```diff
@@ -44,6 +44,13 @@ const SingleBlogView = () => {
       dispatch(getBlogById(id));
   }, [dispatch]);
 
+  const handleDeleteBlog = async (id) => {
+    
+      await dispatch(deleteBlogAction(id));
+      window.location.href = "/blogs#/applicant/blogs";
+    
+  }
```
Comment (ncutixavier) [inline src/pages/Blogs/singleBlog.tsx]: Please improve this implementation, for instance if there is any error this will be wrong

Comment (senseidudu) [reply src/pages/Blogs/singleBlog.tsx]: Error handling is the redux actions plus the toasters

**Comment by MnorbertVii** [comment]
- I think that performing a non-reversible action like this, should include a confirmation step to prevent accidental deletion by the user. 

- try to restrict a non-author user to delete a blog.

File: src/pages/Blogs/singleBlog.tsx [hunk 2]
```diff
@@ -44,6 +44,15 @@ const SingleBlogView = () => {
...
 
+  const handleDeleteBlog = async (id) => {
+    try {
+      await dispatch(deleteBlogAction(id));
+      window.location.href = "/blogs#/applicant/blogs";
+    } catch (error) {
+      console.log(error);
+    }
+  }
+
```
Comment (ncutixavier) [inline src/pages/Blogs/singleBlog.tsx]: - Remove all logs
- This implementation seems to be similar with previous 
- You should find a way to redirect user when request success

Comment (senseidudu) [reply src/pages/Blogs/singleBlog.tsx]: i've fixed all the issues, please review again

**Review by Ishimwe7 — APPROVED** [review]
good job @mutsinziisaac

**Review by MnorbertVii — APPROVED** [review]

**Review by SaddockAime — APPROVED** [review]

**Review by Mugisha146 — APPROVED** [review]

**Review by ncutixavier — APPROVED** [review]

**Pull request merged**